### PR TITLE
[[ Bug 18537 ]] Fix crash when save dialog shown on OSX

### PR DESCRIPTION
**Pull request by livecodeian**

### CODE REVIEW

**Comment by peter-b** [comment]
@livecodeian I think we're going to need this in `develop-8.1`, I'm afraid.

**Comment by livecodeian** [comment]
The bug doesn't occur in 8.1, presumably because it's still getting built with osx10.8 sdk

**Comment by peter-b** [comment]
@livecodeian That would make sense. Interesting!

File: engine/src/mac-dialog.mm [hunk 1]
```diff
@@ -646,12 +646,6 @@ void MCPlatformBeginFileDialog(MCPlatformFileDialogKind p_kind, MCPlatformWindow
 	else 
 		[t_panel setCanCreateDirectories: YES];
 	
-	if ([t_panel respondsToSelector:@selector(isAccessoryViewDisclosed)])
```
Comment (livecodepanos) [inline engine/src/mac-dialog.mm:649]: Don't we need this block for backwards compatibility? Could we instead do something like:

```
if ([t_panel isMemberOfClass:[NSOpenPanel class]] && [t_panel respondsToSelector:@selector(isAccessoryViewDisclosed)])
```

Comment (livecodeian) [reply engine/src/mac-dialog.mm:649]: Hi Panos, I tested this on El Capitan, where the bug was originally reported - removing this code doesn't appear to have any negative effect on the appearance of the accessoryview so i considered it safe to remove.

**Comment by livecodeian** [comment]
@livecode-vulcan try ab27368

**Comment by peter-b** [comment]
@livecode-vulcan review ok 34f0430

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 34f0430c91c16d4795f7c7dbb8b1a134a4ac91d8

**Comment by livecode-vulcan** [comment]
😎 test success 34f0430c91c16d4795f7c7dbb8b1a134a4ac91d8
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/409)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/414)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/409)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/409)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/415)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/409)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/409)

**Pull request merged**